libsigner/src/v0/messages.rs

```diff
@@ -214,6 +231,73 @@ pub trait StacksMessageCodecExtensions: Sized {
     fn inner_consensus_deserialize<R: Read>(fd: &mut R) -> Result<Self, CodecError>;
 }
 
+/// A signer's mock signature across its last seen Stacks Consensus Hash. This is only used
+/// by Epoch 2.5 signers to simulate the signing of a block for every sortition.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct MockSignature {
+    /// The signature across the stacks consensus hash
+    signature: MessageSignature,
+    /// The block hash that the signature is across
+    pub stacks_consensus_hash: ConsensusHash,
+}
+
+impl MockSignature {
+    /// Create a new mock signature with the provided stacks consensus hash and private key
+    pub fn new(
+        stacks_consensus_hash: ConsensusHash,
+        stacks_private_key: &StacksPrivateKey,
+    ) -> Self {
+        let mut sig = Self {
+            signature: MessageSignature::empty(),
+            stacks_consensus_hash,
+        };
+        sig.sign(stacks_private_key)
+            .expect("Failed to sign MockSignature");
+        sig
+    }
+
+    /// The signature hash for the mock signature
+    pub fn signature_hash(&self) -> Result<Sha512Trunc256Sum, CodecError> {
+        let mut hasher = Sha512_256::new();
+        let fd = &mut hasher;
+        write_next(fd, &self.stacks_consensus_hash)?;
+        Ok(Sha512Trunc256Sum::from_hasher(hasher))
+    }
```

Comment:
I think this should use a SIP-18 structured data hash.

Reply:
Oh poo. Forgot about this XD